opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1254,15 +1255,52 @@ namespace detail {
 
     namespace detail
     {
+        /// \brief Compute the L-infinity norm of a vector
+        /// \warn This function is not suitable to compute on the well equations.
+        /// \param a The container to compute the infinity norm on.
+        ///          It has to have one entry for each cell.
+        /// \param info In a parallel this holds the information about the data distribution.
+        double infinityNorm( const ADB& a, const boost::any& pinfo=boost::any() )
+        {
+#if HAVE_MPI
+            if ( pinfo.type() == typeid(ParallelISTLInformation) )
+            {
+                const ParallelISTLInformation& real_info =
+                    boost::any_cast<const ParallelISTLInformation&>(pinfo);
+                double result=0;
+                real_info.computeReduction(a.value(), Reduction::makeGlobalMaxFunctor<double>(), result);
+                return result;
+            }
+            else
+#endif
+            {
+                if( a.value().size() > 0 ) {
+                    return a.value().matrix().lpNorm<Eigen::Infinity> ();
+                }
+                else { // this situation can occur when no wells are present
+                    return 0.0;
+                }
+            }
+        }
 
-        double infinityNorm( const ADB& a )
+        /// \brief Compute the L-infinity norm of a vector representing a well equation.
+        /// \param a The container to compute the infinity norm on.
+        /// \param info In a parallel this holds the information about the data distribution.
+        double infinityNormWell( const ADB& a, const boost::any& pinfo )
         {
+            double result=0;
             if( a.value().size() > 0 ) {
-                return a.value().matrix().lpNorm<Eigen::Infinity> ();
+                result = a.value().matrix().lpNorm<Eigen::Infinity> ();
             }
-            else { // this situation can occur when no wells are present
-                return 0.0;
+#if HAVE_MPI
+            if ( pinfo.type() == typeid(ParallelISTLInformation) )
+            {
+                const ParallelISTLInformation& real_info =
+                    boost::any_cast<const ParallelISTLInformation&>(pinfo);
+                result = real_info.communicator().max(result);
```

Comment:
The function infinityNorm assumes one unknown per cell. Any copy cells (where another process is responsible for computing) are switch off during the local reduction. This switching off might segfault if the vector is smaller than the number of cells of the local grid. 

In infinityNormWell the vector length is only the number of perforated cells. We assume that the whole well is contained on the local grid without any copy cells. Therefore we can just reuse  the local computation and calculate the global maximum, Using infinityNorm for this might segfault.